server/src/main/java/org/apache/iotdb/db/engine/cache/ChunkMetadataCache.java

```diff
@@ -47,45 +47,55 @@
   private static final Logger logger = LoggerFactory.getLogger(ChunkMetadataCache.class);
   private static final IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
   private static final long MEMORY_THRESHOLD_IN_B = config.getAllocateMemoryForChunkMetaDataCache();
-  private static boolean cacheEnable = config.isMetaDataCacheEnable();
+  private static final boolean cacheEnable = config.isMetaDataCacheEnable();
+  private static final long CHUNK_METADATA_FIXED_RAM_SIZE = 160;
+
   /**
    * key: file path dot deviceId dot sensorId.
    * <p>
    * value: chunkMetaData list of one timeseries in the file.
    */
-  private final LRULinkedHashMap<String, List<ChunkMetadata>> lruCache;
+  private final LRULinkedHashMap<AccountableString, List<ChunkMetadata>> lruCache;
 
   private final ReadWriteLock lock = new ReentrantReadWriteLock();
 
-  private AtomicLong cacheHitNum = new AtomicLong();
-  private AtomicLong cacheRequestNum = new AtomicLong();
+  private final AtomicLong cacheHitNum = new AtomicLong();
+  private final AtomicLong cacheRequestNum = new AtomicLong();
 
 
   private ChunkMetadataCache(long memoryThreshold) {
     if (cacheEnable) {
       logger.info("ChunkMetadataCache size = " + memoryThreshold);
     }
-    lruCache = new LRULinkedHashMap<String, List<ChunkMetadata>>(memoryThreshold, true) {
+    lruCache = new LRULinkedHashMap<AccountableString, List<ChunkMetadata>>(memoryThreshold) {
       @Override
-      protected long calEntrySize(String key, List<ChunkMetadata> value) {
+      protected long calEntrySize(AccountableString key, List<ChunkMetadata> value) {
         if (value.isEmpty()) {
-          return key.getBytes().length + averageSize * value.size();
+          return RamUsageEstimator.sizeOf(key) + RamUsageEstimator.shallowSizeOf(value);
         }
+        long entrySize;
         if (count < 10) {
-          long currentSize = RamUsageEstimator.shallowSizeOf(value.get(0)) + RamUsageEstimator
-              .shallowSizeOf(value.get(0).getStatistics());
+          long currentSize = CHUNK_METADATA_FIXED_RAM_SIZE + RamUsageEstimator
```

Comment:
This may be not enough, it also has a statistics, in which different data type may have different occupation, especially the TEXT type.